lib/pf/web/dispatcher.pm

```diff
@@ -74,11 +74,18 @@ sub handler {
         return proxy_redirect($r);
     }
 
-    # Captive-portal detection mecanism bypass
-    if ( ($url =~ /$WEB::CAPTIVE_PORTAL_DETECTION_MECANISM_URLS/o || $user_agent =~ /CaptiveNetworkSupport/s) && isenabled($Config{'captive_portal'}{'detection_mecanism_bypass'}) ) {
+    # Captive-portal detection mecanism
+    # Ability to bypass captive-portal detection mecanism
+    my $captive_portal_detection_mecanism_urls = pf::web::util::build_captive_portal_detection_mecanisms_regex;
+    if ( ($url =~ /$captive_portal_detection_mecanism_urls/o || $user_agent =~ /CaptiveNetworkSupport/s) && isenabled($Config{'captive_portal'}{'detection_mecanism_bypass'}) ) {
         $logger->info("Dealing with a endpoint / browser with captive-portal detection capabilities while having captive-portal detection mecanism bypass enabled. Proxying");
         return proxy_redirect($r);
     }
+    # Enforce HTTP instead of HTTPS when dealing with captive-portal detection mecanism and having a self-signed SSL certificate
+    elsif ( ($url =~ /$captive_portal_detection_mecanism_urls/o || $user_agent =~ /CaptiveNetworkSupport/s) && pf::web::util::is_certificate_self_signed ) {
```

Comment:
Does captive_portal.detection_mecanism_bypass need to be check for self signed also
If so it would be better rewritten like so

```
if ( ($url =~ /$captive_portal_detection_mecanism_urls/o || $user_agent =~ /CaptiveNetworkSupport/s) && isenabled($Config{'captive_portal'}{'detection_mecanism_bypass'}) ) {
    if(pf::web::util::is_certificate_self_signed) {
        $logger->info("Dealing with a endpoint / browser with captive-portal detection capabilities while having a self-signed SSL certificate. Using HTTP instead of HTTPS");
        return html_redirect($r, { secure => $FALSE });  
    } 
    $logger->info("Dealing with a endpoint / browser with captive-portal detection capabilities while having a self-signed SSL certificate. Using HTTP instead of HTTPS");
    return html_redirect($r, { secure => $FALSE });
}
```